### ISSUE TITLE
Repeated creation/deletion of resources breaks cluster

### ISSUE THREAD

**Issue by namliz** [cncf/demo#25]
A successful run from bootstrapping a cluster to provisioning its needed resources all in one go has been happening for some time now. 

With that being said, the development experience is one of tinkering. 

> kubectl create -f
> kubectl delete -f
> kubectl create -f

And so on. Sometimes mysterious bugs pop up, connectivity issues, resources not addressable, time is spent chasing it.. a full cluster shutdown and trying again from scratch often ends up fixing everything. 

The problems are usually around Kubernetes services getting 'confused' or 'sticky'. Additionally, sometimes you see a pod get stuck in 'Terminating' or an inability to attach to running pods because the docker process on the Node crashed. 

So in short, small seemingly harmless actions can currently bring down a whole cluster.

**Comment by namliz** [comment]
Yeah, this is definitely a thing:
kubernetes/kubernetes#25456

Another one to watch out for on top of these is creating a pod with a configmap based volume, before the configmap is created. 

You would not expect a big deal, just add the configmap it wants and it will unstick -- nope, or delete the pod, nope, gets stuck there too.

**Comment by leecalcote** [comment]
@zilman this is good to note and identify as a caveat in our write-ups.

**Comment by namliz** [comment]
@leecalcote  I've had this experience on every combination of cluster I played with long enough. Whether its my bootstrap or the official install, 'etc.

At some point instead of being a human fuzzer it will be time to write one and contribute a test suite. This should definitely be hardened against in future k8s versions.
